### DISCUSSION TITLE
feature: Explicit CUDA, OpenCL support in Clangd extra

### DISCUSSION THREAD

**Discussion by kalocsaibotond** [LazyVim/LazyVim#3472] [Extra Requests]
### Did you check the docs?

- [X] I have read all the LazyVim docs

### Is your feature request related to a problem? Please describe.

Currently, LazyVim with Clangd extra does not know about the existence of CUDA and OpenCL files at all, despite the fact that much of the tool used in the extra (like the Clangd language server) already supports them natively. Neovim assigns lisp filetype for OpenCl files.

At the moment, I circumvented the issue with an autocommand that sets the filetype to c, when I open an OpenCL file. This way I can access much of the features. Though it is a little bit buggy solution (for example there is no syntax highlight for telescope preview).

### Describe the solution you'd like

Explicit support for CUDA, OpenCl, akin to what already is for C / C++.

### Describe alternatives you've considered

LazyVim with the Clang extra, should detect CUDA and OpenC files, and have all the features that it have for C / C++.

### Additional context

Since CUDA and OpenCL are simple extensions of the basic C, many of the Clangd extra tools support them natively. It should not be much difficulty to add them, because only some filetype definition and assignment is needed.

**Comment by dpetka2001** [comment]
The languages Extras come from user contributions mostly who dabble in the specific domain. Feel free to submit a PR.
